File: codeforces/old/temp/justfinish.cpp
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

int main() {
  int T,N,fuel,i,j,x;
  vector<int> trk;

  cin >> T;
  for (int c=1;c<=T;++c) {
    cin >> N;
    trk.clear();
    for (i=0;i<N;++i) {
      cin >> x;
      trk.push_back(x);
    }
    for (i=0;i<N;++i) {
      cin >> x;
      trk[i] -= x;
    }
    fuel = 0;
    i = 0;
    j = 0;
    while (i < N && j < trk.size()) {
      fuel += trk[j];
      while (fuel < 0) {
        fuel -= trk[i];
        trk.push_back(trk[i]);
        i+=1;
      }
      j+=1;
    }
    if (i >= N) {
      cout << "Case " << c << ": Not possible\n";
    }
    else {
      cout << "Case " << c << ": Possible from station " << i+1 << '\n';
    }
  }
  return 0;
}

File: codeforces/old/temp/justfinish.py
from collections import deque

if __name__ == "__main__":
    T = int(input().strip())
    out = []
    for t in range(T):
        N = int(input().strip())
        plist = list(map(int,input().split()))
        qlist = list(map(int,input().split()))
        trk = deque()
        for i in range(N):
            trk.append(plist[i]-qlist[i])
        i,j = 0,0
        fuel = 0
        while i < N and j < len(trk):
            fuel += trk[j]
            while fuel < 0:
                fuel -= trk[i]
                trk.append(trk[i])
                i+=1
            j+=1
        if i >= N:
            out.append("Case {}: Not possible".format(t+1))
        else:
            out.append("Case {}: Possible from station {}".format(t+1,i+1))
    print('\n'.join(out))
